packages/@aws-cdk/aws-appmesh/lib/tls-certificate.ts

```diff
@@ -0,0 +1,192 @@
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import { CfnVirtualGateway, CfnVirtualNode } from './appmesh.generated';
+
+/**
+ * Enum of supported TLS modes
+ */
+export enum TlsMode {
+  /**
+   * Only accept encrypted traffic
+   */
+  STRICT = 'STRICT',
+
+  /**
+   * Accept encrypted and plaintext traffic.
+   */
+  PERMISSIVE = 'PERMISSIVE',
+
+  /**
+   * TLS is disabled, only accept plaintext traffic.
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * Redundant API shapes are represented here.
```

Comment:
Interesting comment 🙂.

Remember that this is a client-facing documentation piece, so it would be good to make it read nicely. Something like "The type returned from {@link TlsCertificate.bind}".